python/cudf/cudf/core/dtypes.py

```diff
@@ -103,3 +105,42 @@ def deserialize(cls, header, frames):
             categories_header, categories_frames
         )
         return cls(categories=categories, ordered=ordered)
+
+
+class ListDtype:
+    def __init__(self, value_type):
+        if isinstance(value_type, ListDtype):
+            self._typ = pa.list_(value_type._typ)
+        else:
+            value_type = cudf.utils.dtypes.np_to_pa_dtype(np.dtype(value_type))
+            self._typ = pa.list_(value_type)
+
+    @property
+    def value_type(self):
+        if isinstance(self._typ.value_type, pa.ListType):
+            return ListDtype.from_arrow(self._typ.value_type)
+        else:
+            return np.dtype(self._typ.value_type.to_pandas_dtype()).name
+
+    @property
+    def name(self):
+        return "list"
+
+    @property
+    def type(self):
+        return list
```

Comment:
Does it makes sense to do this? `type` should return something that can be used to create scalars of this dtype.

Reply:
Either this, or a numpy array that has the item type as its type, that way you'd convert scalars to single element numpy arrays. One vision I had for an eventual `cudf.dtype` was that all of our dtypes would use this api to produce `cudf.Scalar` objects that wrap the equivalent libcudf scalars.

Reply:
+1 to the vision of eventually being able to produce `cudf.Scalar` objects from doing something like `cudf.dtype("int64").type(5)`.

In the meantime, using `list` doesn't do a good job of capturing the dtype as a list can contain arbitrary Python objects whereas we're strongly typed.